pkg/azureclients/vmclient/azure_vmclient.go

```diff
@@ -71,6 +77,7 @@ func New(config *azclients.ClientConfig) *Client {
 		rateLimiterReader: rateLimiterReader,
 		rateLimiterWriter: rateLimiterWriter,
 		subscriptionID:    config.SubscriptionID,
+		cloudName:         config.CloudName,
```

Comment:
why add a new field here, what about use `config.CloudName` directly, refer to

https://github.com/kubernetes-sigs/cloud-provider-azure/blob/083cf6de2feb34b98873082c60b6cd0eaefc925b/pkg/azureclients/diskclient/azure_diskclient.go#L62-L64

Reply:
for consistency, there are 2 clients (for example the interface client) that use an additional API version (the compute api version). In those cases the switch to the Azure Stack API version happens using this field as the config is not available on the method that uses the compute API version. I kept the implementation the same across all clients.

Reply:
all clients are initialized from azure cloud config, what's the benefit to add a redundant field(`cloudName`), only for unit test?

https://github.com/kubernetes-sigs/cloud-provider-azure/blob/74926ac455ea163d9f0a3a7b3f3dd7c4332dc785/pkg/provider/azure.go#L557

Reply:
For unit test yes, and in this case for consistency with other clients like the interface client that actually do need the new field. For example the change in the interface client (already merged) uses the new field to check for the cloud name in a method outside of the initialization.
https://github.com/kubernetes-sigs/cloud-provider-azure/blob/9aae7de0440b8a87e29224ca94e9ef8c47f93e21/pkg/azureclients/interfaceclient/azure_interfaceclient.go#L190

Reply:
thanks for the clarification, lgtm.